packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1447,28 +1428,43 @@ export class ContainerRuntime
 			this.summarizerNode.updateBaseSummaryState(baseSnapshot);
 		}
 
+		const parentContext = cloneParentContext(this);
```

Comment:
Ideally, we should look at using JS proxy here, if it works. The part that is hard - you do not know for sure if a property has a getter or is constant - I've been burned (while building it) with just grabbing current value from a parent, as it might change in the future (like registry for data store context - it resolves only when channel is realized).